src/extensions/gva_roi_metadata.py

```diff
@@ -33,16 +33,16 @@ def process_frame(self, frame):
 
                 for obj in message_objects_array:
                     if "roi_type" in obj:
-                        if obj["roi_type"] == self.roi:
+                        if obj["roi_type"] in self.roi_list:
                             roi_objects.append(obj)
                         else:
                             other_objects.append(obj)
 
                 for obj in other_objects:
-                    for objects in roi_objects:
-                        if (obj["x"] >= objects["x"] and obj["x"] + obj["w"] <= objects["x"] + objects["w"] and
-                                obj["y"] >= objects["y"] and obj["y"] + obj["h"] <= objects["y"] + objects["h"]):
-                            obj["roi_type"] = self.roi
+                    for roi_obj in roi_objects:
+                        if (obj["x"] >= roi_obj["x"] and obj["x"] + obj["w"] <= roi_obj["x"] + roi_obj["w"] and
+                                obj["y"] >= roi_obj["y"] and obj["y"] + obj["h"] <= roi_obj["y"] + roi_obj["h"]):
```

Comment:
for code readability, i would suggest putting this as a named separate function (e.g. is_object_in_bagging_roi() or something meaningful).